File: sdks/csharp/sdk/src/PogodocApi/Tokens/TokensClient.cs
using System.Net.Http;
using System.Threading;
using global::System.Threading.Tasks;
using PogodocApi.Core;

namespace PogodocApi;

public partial class TokensClient
{
    private RawClient _client;

    internal TokensClient(RawClient client)
    {
        _client = client;
    }

    /// <summary>
    /// Invalidates an API token by storing it in the deleted tokens S3 bucket, preventing future use of the token for authentication.
    /// </summary>
    /// <example><code>
    /// await client.Tokens.DeleteApiTokenAsync("tokenId");
    /// </code></example>
    public async global::System.Threading.Tasks.Task DeleteApiTokenAsync(
        string tokenId,
        RequestOptions? options = null,
        CancellationToken cancellationToken = default
    )
    {
        var response = await _client
            .SendRequestAsync(
                new JsonRequest
                {
                    BaseUrl = _client.Options.BaseUrl,
                    Method = HttpMethod.Delete,
                    Path = string.Format(
                        "api-tokens/{0}",
                        ValueConvert.ToPathParameterString(tokenId)
                    ),
                    Options = options,
                },
                cancellationToken
            )
            .ConfigureAwait(false);
        if (response.StatusCode is >= 200 and < 400)
        {
            return;
        }
        {
            var responseBody = await response.Raw.Content.ReadAsStringAsync();
            throw new PogodocApiApiException(
                $"Error with status code {response.StatusCode}",
                response.StatusCode,
                responseBody
            );
        }
    }
}

File: sdks/php/src/sdk/src/Tokens/TokensClient.php
<?php

namespace Pogodoc\Tokens;

use GuzzleHttp\ClientInterface;
use Pogodoc\Core\Client\RawClient;
use Pogodoc\Exceptions\PogodocException;
use Pogodoc\Exceptions\PogodocApiException;
use Pogodoc\Core\Json\JsonApiRequest;
use Pogodoc\Environments;
use Pogodoc\Core\Client\HttpMethod;
use GuzzleHttp\Exception\RequestException;
use Psr\Http\Client\ClientExceptionInterface;

class TokensClient
{
    /**
     * @var array{
     *   baseUrl?: string,
     *   client?: ClientInterface,
     *   maxRetries?: int,
     *   timeout?: float,
     *   headers?: array<string, string>,
     * } $options
     */
    private array $options;

    /**
     * @var RawClient $client
     */
    private RawClient $client;

    /**
     * @param RawClient $client
     * @param ?array{
     *   baseUrl?: string,
     *   client?: ClientInterface,
     *   maxRetries?: int,
     *   timeout?: float,
     *   headers?: array<string, string>,
     * } $options
     */
    public function __construct(
        RawClient $client,
        ?array $options = null,
    ) {
        $this->client = $client;
        $this->options = $options ?? [];
    }

    /**
     * Invalidates an API token by storing it in the deleted tokens S3 bucket, preventing future use of the token for authentication.
     *
     * @param string $tokenId
     * @param ?array{
     *   baseUrl?: string,
     *   maxRetries?: int,
     *   timeout?: float,
     *   headers?: array<string, string>,
     *   queryParameters?: array<string, mixed>,
     *   bodyProperties?: array<string, mixed>,
     * } $options
     * @throws PogodocException
     * @throws PogodocApiException
     */
    public function deleteApiToken(string $tokenId, ?array $options = null): void
    {
        $options = array_merge($this->options, $options ?? []);
        try {
            $response = $this->client->sendRequest(
                new JsonApiRequest(
                    baseUrl: $options['baseUrl'] ?? $this->client->options['baseUrl'] ?? Environments::Default_->value,
                    path: "api-tokens/{$tokenId}",
                    method: HttpMethod::DELETE,
                ),
                $options,
            );
            $statusCode = $response->getStatusCode();
            if ($statusCode >= 200 && $statusCode < 400) {
                return;
            }
        } catch (RequestException $e) {
            $response = $e->getResponse();
            if ($response === null) {
                throw new PogodocException(message: $e->getMessage(), previous: $e);
            }
            throw new PogodocApiException(
                message: "API request failed",
                statusCode: $response->getStatusCode(),
                body: $response->getBody()->getContents(),
            );
        } catch (ClientExceptionInterface $e) {
            throw new PogodocException(message: $e->getMessage(), previous: $e);
        }
        throw new PogodocApiException(
            message: 'API request failed',
            statusCode: $statusCode,
            body: $response->getBody()->getContents(),
        );
    }
}
